### PR TITLE
(DOCSP-29237): Kotlin FAQ

### PR DESCRIPTION
**Pull request by mongodben**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29237
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/DOCSP-29237/faq/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -75,13 +76,12 @@ and links to the API documentation.
    * - `Kotlin Driver <{+api-kotlin+}/apidocs/mongodb-driver-kotlin-coroutine/index.html>`__
      - API
 
-.. TODO:(DOCSP-29237)
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/index.txt:78]: done!

File: source/faq.txt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+   :depth: 2
+   :class: singlecol
+
+How Is the Kotlin Driver Different from Kmongo?
+-----------------------------------------------
+
+The Kotlin driver is the official MongoDB driver for Kotlin. It is
+developed by the MongoDB team and provides a native API for Kotlin
+applications to connect to MongoDB and work with data. It is implemented
+by wrapping the MongoDB Java driver.
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/faq.txt:19]: done

File: source/faq.txt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+The official Kotlin driver and Kmongo generally have similar APIs.
+Notable similarities between the Kotlin driver and Kmongo include:
+
+- Support for synchronous and coroutine-based operations
+- Support using data classes to represent MongoDB documents
+- Support KotlinX serialization
+- Full support for MongoDB CRUD APIs and aggregation
+
+The official Kotlin driver and Kmongo generally have similar APIs, but there are some
+key differences:
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/faq.txt:39]: idk about plans for changes

File: source/faq.txt [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+- The official driver does not support MongoDB shell commands.
+- The official driver support type-sage queries with the Builders API,
+  whereas Kmongo uses infix functions and property references for type-safe queries.
+
+.. _kotlin-faq-connection-pool:
+
+How Does Connection Pooling Work in the Kotlin Driver?
+------------------------------------------------------
+
+Every ``MongoClient`` instance has a built-in connection pool for each server
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/faq.txt:53]: added a link at the end of the section

File: source/faq.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashni-mongodb) [reply source/faq.txt:39]: Jackson's something we've been thinking about and we do have [JAVA-2483](https://jira.mongodb.org/browse/JAVA-2483) in the backlog. Not the case for Gson/RxJava though.

File: source/faq.txt [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+applications to connect to MongoDB and work with data. It is implemented
+by wrapping the :driver:`MongoDB Java driver </java-drivers/>`.
+
+`KMongo <https://litote.org/kmongo/>`__ is a popular community-developed library
+for working with MongoDB from Kotlin applications.
+It is also a wrapper around the Java driver.
+It was created prior to the creation of the official Kotlin driver to serve
+the needs of the Kotlin community.
+
+The Kotlin driver was developed in collaboration with the creator of KMongo,
```
Comment (ashni-mongodb) [inline source/faq.txt:27]: Maybe something to explain why we built this when Kmongo already existed -- like "The MongoDB server-side Kotlin driver was developed to give users an officially-supported driver" or something like that?

**Review by ashni-mongodb — APPROVED** [review]
left a comment but otherwise LGTM, thank you!

File: examples/src/test/kotlin/CodecTest.kt [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+
+// :replace-start: {
+//    "terms": {
+//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string uri>\""
+//    }
+// }
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class CodecTest {
```
Comment (jyemin) [inline examples/src/test/kotlin/CodecTest.kt:26]: Is this intended to be reviewed?  It's not referenced in the FAQ.

Comment (mongodben) [reply examples/src/test/kotlin/CodecTest.kt:26]: ohh, this is a git error on my part. i'll delete that from the PR. thx for spotting.

Comment (mongodben) [reply examples/src/test/kotlin/CodecTest.kt]: fixed

**Review by jyemin — APPROVED** [review]

**Review by jyemin — APPROVED** [review]

**Review by jyemin — APPROVED** [review]

**Pull request merged**